### DISCUSSION TITLE
Saving search_data during the optimization process

### DISCUSSION THREAD

**Discussion by dthillaithevan** [SimonBlanke/Gradient-Free-Optimizers#45] [Q&A] [answered]
Hi,

Is there anyway to save intermediate results (i.e. data contained within `opt.search_data`) each iteration? My motivation for doing for doing this is to cover the scenario where the optimization is terminated for what ever reason, I can simply load the previous results using `memory_warm_start` without needing to start the optimization from scratch. 

Thanks!

**Comment by SimonBlanke** [comment, marked as answer]
Hello @dthillaithevan,

This is a common request! :-) I wrote the [search-data-collector](https://github.com/SimonBlanke/search-data-collector) for this problem. It can write the search-data in each iteration into csv-files (also works in parallel) or into an sqlite database (new feature, no examples yet).

An example for Gradient-Free-Optimizers (with csv-files) can be found here:
https://github.com/SimonBlanke/search-data-collector/blob/main/examples/append_data_gradient_free_optimizers.py

The search-data can be passed into the next optimization run without any changes.

Let me know if you have more questions about this.

**Reply by dthillaithevan** [reply]
Thanks for the speedy reply! This is exactly what I was looking for. 
Best,
Dilaksan
